### PR TITLE
Return insufficient memory error when load failed

### PR DESCRIPTION
**Pull request by bigsheeper**
Signed-off-by: bigsheeper <yihao.dai@zilliz.com>

issues: https://github.com/milvus-io/milvus/issues/21573

### CODE REVIEW

**Review by yah01 — CHANGES_REQUESTED** [review]
Need to remove the previous load status when received a load request

**Comment by bigsheeper** [comment]
> Need to remove the previous load status when received a load request

Thanks for reminding

**Review by yah01 — COMMENTED** [review]
FYI: now `LoadSegments` tries best-effort to load segments, which means it may only load   some of segments of requests, and returns error

File: internal/querycoordv2/meta/load_status_cache.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+const LoadStatusExpireTime = 24 * time.Hour
+
+var GlobalLoadCache *LoadStatusCache
+
+type LoadStatus struct {
+	s *commonpb.Status
+	t time.Time
+}
+
+type LoadStatusCache struct {
```
Comment (yah01) [inline internal/querycoordv2/meta/load_status_cache.go:40]: Name the methods to `Get`, `Put`, `Remove` and `TryExpire` make more readability for me

**Comment by yah01** [comment]
The collection observer needs to check whether LoadCache contains error for every collection, and if so, cancel the loading collection

File: internal/querycoordv2/task/scheduler.go [hunk 2]
```diff
@@ -658,6 +664,33 @@ func (scheduler *taskScheduler) RemoveByNode(node int64) {
 	}
 }
 
+func (scheduler *taskScheduler) recordSegmentTaskError(index replicaSegmentIndex, task *SegmentTask) {
+	// for now, we only record insufficient memory error
+	if !errors.Is(task.Err(), ErrInsufficientMemory) {
+		return
```
[the remark this answers is not in the document]
Comment (bigsheeper) [reply internal/querycoordv2/task/scheduler.go:670]: lgtm

**Comment by bigsheeper** [comment]
> The collection observer needs to check whether LoadCache contains error for every collection, and if so, cancel the loading collection

You're right, I'll add a new observe behavior

**Review by yah01 — APPROVED** [review]
/lgtm

File: internal/querycoordv2/meta/load_status_cache.go (hunk 1 above)
Comment (jaime0815) [inline internal/querycoordv2/meta/load_status_cache.go:40]: You can use the local cache to replace this implementation.
https://github.com/milvus-io/milvus/blob/9fd9d9cc05236721e5415a9234d942e91fd48568/internal/util/cache/local_cache.go#L45

Comment (bigsheeper) [reply internal/querycoordv2/meta/load_status_cache.go:40]: By the way, dose the local cache support to expire cache if out of date? @jaime0815

Comment (yah01) [reply internal/querycoordv2/meta/load_status_cache.go:40]: local cache is not thread-safe now, see https://github.com/milvus-io/milvus/issues/21605

**Comment by yah01** [comment]
The error should be not visible before canceling the load job, or ShowCollection will get error while loading

**Pull request merged**